easybuild/easyblocks/c/clang.py

```diff
@@ -237,6 +242,15 @@ def configure_step(self):
         else:
             self.cfg['configopts'] += "-DLLVM_ENABLE_ASSERTIONS=OFF "
 
+        if self.cfg["build_lld"]:
+            self.cfg['configopts'] += "-DLLVM_ENABLE_PROJECTS=lld "
+
+        # If CUDA is included as a dep, add it as a target
+        if get_software_root("CUDA") and "NVPTX" not in self.cfg['build_targets']:
+            self.cfg['build_targets'] += ["NVPTX"]
```

Comment:
Use `self.cfg.update`, this most likely doesn't have the intended effect of adding `NVPTX` to the `build_targets` list...

The problem is that `self.cfg['build_targets']` returns a *copy* of the value (with templates resolved), which is why you have to use `self.cfg.update`

Reply:
It does indeed work but I don't mind changing it

Reply:
Hmm, ok, I'm probably confusing with a similar thing that doesn't work as expected.

I can confirm that the existing `+=` statements for e.g. `-DCMAKE_BUILD_TYPE` work as intended, but using `self.cfg.update` makes more sense.